src/Approximations/overapproximate.jl

```diff
@@ -516,84 +516,103 @@ function overapproximate(cap::Intersection{N,
+                         ::Type{CartesianProductArray{N, S}}
+                        ) where {N, S<:LazySet{N}}
+    cpa = array(lm.X)
+    arr = Vector{S}(undef, length(cpa))
+    return _overapproximate_lm_cpa!(arr, lm.M, cpa, S)
 end
 
 """
-    overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                      ::Type{CartesianProductArray{N, HPolytope{N}}},
-                   dir::AbstractDirections{N}) where {N, T<:LazySet{N}}
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
+                    ::Type{<:CartesianProductArray},
+                    dir::Type{<:AbstractDirections}) where {N}
 
-Decompose a lazy linear map of a cartesian product array.
+Decompose a lazy linear map of a cartesian product array with template
+directions while keeping the original block structure.
 
 ### Input
 
-- `lm`   -- lazy linear map of cartesian product array
+- `lm`  -- lazy linear map of a cartesian product array
 - `CartesianProductArray` -- type for dispatch
-- `dir` -- approximation option for decomposition
+- `dir` -- template directions for overapproximation
 
 ### Output
 
 A `CartesianProductArray` representing the decomposed linear map.
 """
-function overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                           ::Type{<:CartesianProductArray},
-                         dir::Type{<:AbstractDirections}) where {N, T<:LazySet{N}}
-    M, cpa = lm.M, lm.X
-    array = Vector{HPolytope{N}}(undef, length(cpa.array))
-    return _overapproximate_lm_cpa!(M, cpa, array, dir)
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
+                         ::Type{<:CartesianProductArray},
+                         dir::Type{<:AbstractDirections}) where {N}
+    cpa = array(lm.X)
+    arr = Vector{HPolytope{N}}(undef, length(cpa))
+    return _overapproximate_lm_cpa!(arr, lm.M, cpa, dir)
 end
 
-#same for <:AbstractHyperrectangle
-function overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                           ::Type{<:CartesianProductArray},
-                         oa::Type{<:AbstractHyperrectangle}) where {N, T<:LazySet{N}}
-    M, cpa = lm.M, lm.X
-    array = Vector{oa{N}}(undef, length(cpa.array))
-    return _overapproximate_lm_cpa!(M, cpa, array, oa)
+"""
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
+                    ::Type{<:CartesianProductArray},
+                    set_type::Type{<:LazySet}) where {N}
+
+Decompose a lazy linear map of a cartesian product array with a given set type
+while keeping the original block structure.
+
+### Input
+
+- `lm`  -- lazy linear map of a cartesian product array
+- `CartesianProductArray` -- type for dispatch
+- `set_type` -- set type for overapproximation
+
+### Output
+
+A `CartesianProductArray` representing the decomposed linear map.
+"""
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
```

Comment:
This function is a repetition of https://github.com/JuliaReach/LazySets.jl/pull/1425/files#diff-eb4fb67940b1678df4462dad2b59a25cR535 , with a different signature (sounds like it's not a good idea to have many of these methods, but OK), i suggest to add a comment of this fact in the docstring.

Reply:
Again, I did not change functionality here.

> sounds like it's not a good idea to have many of these methods

This is the Julia way, I'm sorry.

Reply:
It doesn't prevent from improving the docstring (same for previous comment). In any case, i opened https://github.com/JuliaReach/LazySets.jl/issues/1431 as follow-up.